src/main/java/com/disi/TravelPoints/service/WishlistService.java

```diff
@@ -0,0 +1,40 @@
+package com.disi.TravelPoints.service;
+
+import com.disi.TravelPoints.dto.LandmarkDetails;
+import com.disi.TravelPoints.dto.UserDetails;
+import com.disi.TravelPoints.model.Landmark;
+import com.disi.TravelPoints.model.Wishlist;
+import com.disi.TravelPoints.repository.WishlistRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class WishlistService {
+    private final WishlistRepository wishlistRepository;
+
+    public List<LandmarkDetails> getWishlistForUser(Long id) {
+        Wishlist wishlist = wishlistRepository.findByUser_Id(id);
+
+        return wishlist.getLandmarks().stream()
```

Comment:
`getLandmarks` throw NullPointerException if wishlist is null. You can modify `findByUser_Id` to return `Optional<Wishlist>`  and use `orElseThrow` method to throw a customException with specific message for UI

Reply:
Yes, you are right